### PR TITLE
[#9175] Remove unnecessary integrity check for getFeedbackQuestionsForSession

### LINKED ISSUE #9175: Remove unnecessary integrity check for getFeedbackQuestionsForSession
https://github.com/TEAMMATES/teammates/blob/92b7744ffafb7162535eacde0051904cdff13659/src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java#L126-L145

The check for the existence of the session is not necessary:

- Extra database cost + latency
- No harm if the course does not exist -> return empty list
- We don't do the parent entity existence check in `session`, `response`, `responseComment` logic also

Aside from the code changes, two more things need to be done along with the PR:

- Update `API for retrieving entities:` section in `design.md` to reflect the new retrieving entity policy.
- Update tests introduced in #9153 to check the non-existence of questions after cascade deletion.

### PR DESCRIPTION
**Pull request by IAmPramod**
Fixes #9175

### CODE REVIEW

**Comment by IAmPramod** [comment]
@xpdavid, Build is failing at Line#464 while creating Feedback Question using session which doesn't have feedback questions.
I can create a copy of existing feedback session and test delete on the same but I think that is already covered as part of method `testCopyFeedbackSession` at Line#471.

File: src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java [hunk 1]
```diff
@@ -60,11 +60,8 @@ public void createFeedbackQuestion(FeedbackQuestionAttributes fqa)
 
         String feedbackSessionName = fqa.feedbackSessionName;
         String courseId = fqa.courseId;
-        List<FeedbackQuestionAttributes> questions = null;
-
-        try {
-            questions = getFeedbackQuestionsForSession(feedbackSessionName, courseId);
-        } catch (EntityDoesNotExistException e) {
+        List<FeedbackQuestionAttributes> questions = getFeedbackQuestionsForSession(feedbackSessionName, courseId);
+        if (questions.isEmpty()) {
```
Comment (xpdavid) [inline src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java]: Empty question list does not mean that the session doesn't exist.

Should use `fsLogic.getFeedbackSession` to check if session exist.

Comment (IAmPramod) [reply src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java]: I have fixed it. I see build failed because several test failing.

**Comment by IAmPramod** [comment]
@xpdavid Ready for review

File: src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java (hunk 1 above)
Comment (IAmPramod) [reply src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java]: Made the change.

**Comment by IAmPramod** [comment]
@xpdavid Gentle Reminder!

File: src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java [hunk 2]
```diff
@@ -654,24 +640,4 @@ private void shiftQuestionNumbersDown(int questionNumberToShiftFrom,
         }
     }
 
-    /*
-     * Removes questions with no recipients.
-     */
-    public List<FeedbackQuestionAttributes> getQuestionsWithRecipients(
```
Comment (xpdavid) [inline src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java]: Is this unused method?

File: src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java [hunk 3]
```diff
@@ -629,13 +617,11 @@ private void deleteFeedbackQuestionCascade(
         // Cascade delete responses for question.
         frLogic.deleteFeedbackResponsesForQuestionAndCascade(questionToDelete.getId(), true);
 
-        List<FeedbackQuestionAttributes> questionsToShiftQnNumber = null;
-        try {
-            questionsToShiftQnNumber = getFeedbackQuestionsForSession(feedbackSessionName, courseId);
-        } catch (EntityDoesNotExistException e) {
+        List<FeedbackQuestionAttributes> questionsToShiftQnNumber =
+                getFeedbackQuestionsForSession(feedbackSessionName, courseId);
+        if (fsLogic.getFeedbackSession(feedbackSessionName, courseId) == null) {
```
Comment (xpdavid) [inline src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java]: The check should do before `getFeedbackQuestionsForSession` ? Same for other place?

File: src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java (hunk 2 above)
Comment (IAmPramod) [reply src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java]: Yes, It was unused.

File: src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java (hunk 3 above)
Comment (IAmPramod) [reply src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java]: Yes. Moved them before questions for session retrieval

**Comment by IAmPramod** [comment]
@xpdavid ready for review.

**Pull request merged**